src/core/utils.js

```diff
@@ -315,22 +315,64 @@ export function removeReSpec(doc) {
 
 /**
  * Adds error class to each element while emitting a warning
- * @param {Element|Array:Elements} elems
+ * @param {Element|Element[]} elems
  * @param {String} msg message to show in warning
  * @param {String} title error message to add on each element
  */
-export function showInlineError(elems, msg, title) {
+export function showInlineWarning(elems, msg, title) {
   if (!Array.isArray(elems)) elems = [elems];
-  if (!elems.length) return;
-  if (!title) title = msg;
-  elems.forEach(elem => {
-    elem.classList.add("respec-offending-element");
-    elem.setAttribute("title", title);
-  });
-  pub("warn", msg + " See developer console for details.");
+  const links = elems
+    .map((element, i) => {
+      markAsOffending(element, msg, title);
+      return generateMarkdownLink(element, i);
+    })
+    .join(", ");
+  pub("warn", msg + ` at: ${links}.`);
   console.warn(msg, elems);
 }
 
+/**
+ * Adds error class to each element while emitting a warning
+ * @param {Element|Element[]} elems
+ * @param {String} msg message to show in warning
+ * @param {String} title error message to add on each element
+ */
+export function showInlineError(elems, msg, title) {
+  if (!Array.isArray(elems)) elems = [elems];
+  const links = elems
+    .map((element, i) => {
+      markAsOffending(element, msg, title);
+      return generateMarkdownLink(element, i);
+    })
+    .join(", ");
+  pub("error", msg + ` at: ${links}.`);
+  console.error(msg, elems);
+}
+
+/**
+ * Adds error class to each element while emitting a warning
+ * @param {Element} elem
+ * @param {String} msg message to show in warning
+ * @param {String} title error message to add on each element
+ */
+function markAsOffending(elem, msg, title) {
+  elem.classList.add("respec-offending-element");
+  if (!elem.hasAttribute("title")) {
+    elem.setAttribute("title", title || msg);
+  }
+  if (!elem.id) {
+    addId(elem, "respec-offender");
+  }
+}
+
+/**
+ * @param {Element} element
+ * @param {number} i
+ */
+function generateMarkdownLink(element, i) {
+  return `[${i + 1}](#${element.id})`;
```

Comment:
I'm not feeling very good about adding just numbers here (`i + 1`), even though I can't think of anything better.

Reply:
Number are ok.

Reply:
>I can't think of anything better.

Neither can I. 😀